org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java

```diff
@@ -2141,10 +2141,10 @@ private Expression convertLiteral(JCLiteral literal) {
 		if (value instanceof Number) {
 			// to check if the literal is actually a prefix expression of it is a hex
 			// negative value we need to check the source char value.
-			char firstChar = this.rawText.substring(literal.getStartPosition(), literal.getStartPosition() + 1)
-					.charAt(0);
-
-			if( firstChar != '-' ) {
+			// fix ASTConverterAST8Test.test0049()/test0052()
+			boolean isNumberLiteral = (this.rawText.substring(literal.getStartPosition(), literal.getStartPosition() + 1)
+					.charAt(0) != '-') || ((value instanceof Integer i) && i == Integer.MIN_VALUE) || ((value instanceof Long l) && l == Long.MIN_VALUE);
```

Comment:
By the way, let's get rid of creating a `substring` here and just check `this.rawText.charAt(literal.getStartPosition() == '-'`

Reply:
I have updated the PR.